backend.native/tests/runtime/workers/freeze2.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package runtime.workers.freeze2
+
+import kotlin.test.*
+
+import konan.worker.*
+
+data class Data(var int: Int)
+
+/* An awful hack: we mark data as immutable, while it is not. */
+@konan.internal.Immutable
+data class SharedData(var field: Any)
+
+@Test fun runTest() {
+    // Ensure that we can not mutate frozen objects and arrays.
+    val a0 = Data(2)
+    var ok = false
+    a0.int++
+    a0.freeze()
+    try {
+        a0.int++
+    } catch (e: InvalidMutabilityException) {
+        ok = true
+    }
+    if (!ok) throw Error("can mutate frozen object: $a0")
+
+    ok = false
+    val a1 = ByteArray(2)
+    a1[1]++
+    a1.freeze()
+    try {
+        a1[1]++
+    } catch (e: InvalidMutabilityException) {
+        ok = true
+    }
+    if (!ok) throw Error("can mutate frozen ByteArray")
```

Comment:
We have some `kotlin.test.*` methods to be used in this case, e.g. [`assertFails`](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-fails.html) and [`assertFailsWith`](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-fails-with.html).

Reply:
Done